---
title: "100-raw-data"
output: html_document
date: "2025-08-22"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(grid)
library(ggpubr)
```


Download data


export CKAN_API_TOKEN=

executing bash script


## Align to reference genome 

in data/align


ls | grep R1 | perl -pe 's/.fastq.gz//g' > f1
ls | grep R2 | perl -pe 's/.fastq.gz//g' > r1
ls | grep R1 | perl -pe 's/_FISH.*.fastq.gz//g' > n1
paste f1 r1 n1 > toalign1.txt
29 fish

toalign2.txt
17 fish

paste f1 r1 n1 > toalign3.txt
2 fish

align3 checksums validated


bash ../../doAlign-zipped.sh toalign1.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz

bash ../../doAlign-zipped.sh toalign2.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz

bash ../../doAlign-zipped.sh toalign3.txt /home/maccamp/lates-wgs/genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna.gz

note: calculate checksums in data/align for sanity

for f in *.fastq.gz; do md5sum $f >> align1.checksums; done;
for f in *.fastq.gz; do md5sum $f >> align2.checksums; done;

### Checksums

```{r}
md5<-read_delim("data/bpa_f00df7f7_20250822T0455/tmp/bpa_f00df7f7_20250822T0455_md5sum.txt", delim="  ", col_names = c("md5","file"))

align1<-read_delim("outputs/100/align1.checksums", delim="  ", col_names = c("md5","file"))
align2<-read_delim("outputs/100/align2.checksums", delim="  ", col_names = c("md5","file"))

mds<-bind_rows(align1, align2) %>% rename(Found=md5)

left_join(md5, mds) %>% filter(md5 != Found)
```
compares to align1.checksums, align2.checksums, all files complete


#failures
616649
616651
616653
616655

# Check metrics

```{r}
m<-read_csv("meta/meta08222025.csv")
stats<-read_csv("outputs/100/align1.csv", col_names = c("library-id","Aligned","Dedup","Coverage"))
stats2<-read_csv("outputs/100/temp.csv", col_names = c("library-id","Aligned","Dedup","Coverage"))
stats3<-read_csv("outputs/100/align3.csv", col_names = c("library-id","Aligned","Dedup","Coverage"))
stats<-bind_rows(stats,stats2,stats3)
m<-left_join(m,stats) %>% mutate(Pheno=ifelse(Phenotype=="SSS",0,1)) %>% mutate(Path=paste0("/home/maccamp/barra-lh/data/bams/",`library-id`,".sort.flt.bam"))
median(na.omit(m$Coverage))
```

```{r}
ggplot(m) + 
  geom_histogram(aes(x=Coverage,fill=Phenotype), alpha=0.75) +
  ylab("Count\n") +
  xlab("\nCoverage after Filtering") +
  theme_bw() +
  scale_fill_manual(values=c("red","blue"))
ggsave("outputs/100/coverage.jpeg")
```
```{r}
m %>% filter(Coverage>10) %>% group_by(Pheno, Phenotype) %>% summarize(Count=n(), MedianCov=median(Coverage), AverageCov=mean(Coverage))
m %>% filter(Coverage>10) %>% select(Path) %>% write_tsv("bamlists/test44.bamlist", col_names = FALSE)
m %>% filter(Coverage>10) %>% select(Pheno) %>% write_tsv("bamlists/test44.pheno", col_names = FALSE)
m44<-m %>% filter(Coverage>10)
meta<-m44
```
SFLc0592 is the low baller here
can put sort.flt.bams and indices in data/bams

# Run a basic PCA using chrom01

````{sh, eval=FALSE}
srun -p high -t 10:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/test44.bamlist -r NC_066833.1 \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 40 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/100/44-ibs-90-Lcal01 > outputs/100/44-ibs-90-Lcal01.out 2> outputs/100/44-ibs-90-Lcal01.err &

```

srun -p high -t 16:00:00 --mem=32G --nodes=1 --cpus-per-task=6  $HOME/angsd/angsd -P 6  \
-bam bamlists/48.bamlist -rf meta/lates-lgs.txt \
-ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-minInd 36 -minMapQ 20 -minQ 20 -GL 2 -doGLF 2 \
-doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doIBS 1 -doCounts 1 -doCov 1 -makeMatrix 1 -minMaf 0.05 \
-out outputs/1304/48-ibs-90 > outputs/1304/48-ibs-90.out 2> outputs/1304/48-ibs-90.err &


```{r}
m <- as.matrix(read.table("outputs/100/44-ibs-90-Lcal01.covMat"))
eig <- eigen(m)
var<-eig$values/sum(eig$values)
cumvar<-cumsum(eig$values)/sum(eig$values)

head(var)
head(cumvar)

vdf<-var %>% as_tibble() %>% rename(Variance=value) %>% mutate(PC=1:n()) %>% head(n=10)

scree<-ggplot(vdf) +
  geom_line(aes(x=PC, y=Variance)) +
  theme_bw() +
  theme(panel.grid = element_blank()) +
  scale_x_continuous(breaks=vdf$PC) +
  theme(axis.text=element_text(size=8)) +
  ggtitle("Scree Plot") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

scree
ggsave("outputs/100/scree-plot.jpeg")
```


```{r}
covs<-eig$vectors[,1:3] %>% as_tibble() %>% bind_cols(meta)

pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Phenotype), pch=21, alpha=0.75, cex=2) +
  #  geom_text(aes(x=V1, y=V2, label=Run), pch=21, alpha=0.75, cex=2) +
   xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  scale_fill_manual(values=c("red","blue")) +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  ggtitle("PCs 1 and 2") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

pc12
ggsave("outputs/100/pc12.jpeg")
```


Must have a few wayward souls

```{r}
pc12<-ggplot(covs) +
  geom_point(aes(x=V1, y=V2, fill=Coverage), pch=21, alpha=0.75, cex=2) +
  #  geom_text(aes(x=V1, y=V2, label=Run), pch=21, alpha=0.75, cex=2) +
   xlab(paste("PC1", " ", round((100*var[1]),2), "%", sep = "")) +
  ylab(paste("PC2", " ", round((100*var[2]),2), "%", sep = "")) +
  scale_fill_viridis_c() +
  theme_bw() +
  theme(panel.grid=element_blank()) +
  ggtitle("PCs 1 and 2") +
  theme(plot.title = element_text(hjust=0.5, face="bold"))

pc12
```