---
title: "102-with-Taiwan"
output: html_document
date: "2025-08-26"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message=FALSE)
```

```{r}
library(tidyverse)
library(RColorBrewer)
library(ggrepel)
library(ggpubr)
```

Most of these fish are catadromous, that is:

This fish was not analyzed:
WJC9188
&
WJC9556 - aquaculture escapee

```{r}
t<-read_csv("~/github/barra-lh/meta/taiwanese-fish.csv") %>% mutate(Path=paste0("/home/maccamp/lates-wgs/data/taiwan/",sample_id,".sort.flt.bam")) %>% filter(!library_id %in% c("WJC9188","WJC9556")) %>% select(-sample_id)
t
```
```{r}
m52<-read_csv("~/github/barra-lh/meta/m52.csv")

ggplot(m52) +
  geom_histogram(aes(x=Coverage, fill=Phenotype))
```
```{r}
m56<-m52 %>% bind_rows(t) %>% filter(Coverage > 10)
m56 %>% group_by(Phenotype) %>% summarize(Count=n())

m56 %>% select(Path) %>% write_tsv("~/github/barra-lh/bamlists/test56.bamlist", col_names = FALSE)
m56 %>% select(Pheno) %>% write_tsv("~/github/barra-lh/bamlists/test56.pheno", col_names = FALSE)

```


```{sh, eval=FALSE}
bash $HOME/barra-lh/101.1-do-asso.sh $HOME/barra-lh/bamlists/test56.bamlist $HOME/barra-lh/meta/lates-lgs.txt  $HOME/barra-lh/bamlists/test56.pheno
```

```{r}
list<-list.files(path = "outputs/102", pattern="lrt0.gz", full.names = TRUE)
lrts<-lapply(list, read_tsv) %>% bind_rows() %>% mutate(log10p = -log10(dchisq(LRT, df = 1))) %>%
  filter(log10p > 1)
```


```{r}
ggplot(lrts) + geom_point(aes(x=Position, y=log10p, color=log10p), alpha=0.75, cex=0.7)+
  geom_hline(yintercept = 6, lty=2, alpha=0.6) +
  scale_color_gradient(low="grey",high="skyblue") +
  theme_bw()+
  theme(axis.text.x= element_text(angle=45,hjust=1)) +
  theme(panel.grid = element_blank()) +
  ylab("-log10(p)") +
  ggtitle("SFF vs SSS GWAS") +
  facet_wrap(.~Chromosome, scales = "free_x") +
  theme(plot.title = element_text(hjust=0.5))

ggsave("outputs/102/qld-taiwan-gwas.jpeg", width=14, height=14)
```
