---
title: "104-dropping-odd-fish"
output: html_document
date: "2025-09-03"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning=FALSE, message = FALSE)
```

Dropping three odd fish and changing Lc0427 to SFF



srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12 \
 -doAsso 1 -yBin $HOME/barra-lh/bamlists/test47.pheno -GL 1 -minInd 45 -minMaf 0.10 \
 -minMapQ 20 -minQ 20 \
 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -rf meta/lates-lgs.txt -out outputs/101/combined-asso-95 \
 -bam $HOME/barra-lh/bamlists/test47.bamlist  > outputs/101/combined-asso.out 2> outputs/101/combined-asso.err &

# Check metrics

```{r}
m<-read_csv(file="meta/m52.csv") %>% filter(!specimen_id %in% c("SFLc0316","SFLc0625","SFLc0592")) %>%
  mutate(NewPheno=ifelse(specimen_id=="SFLc0427", "1",Pheno)) %>% mutate(NewPhenotype=ifelse(specimen_id=="SFLc0427", "SFF",Phenotype))
m  %>% filter(specimen_id=="SFLc0427")
```


```{r}
m %>% filter(Coverage>8) %>% group_by(Pheno, Phenotype) %>% summarize(Count=n(), MedianCov=median(Coverage), AverageCov=mean(Coverage))
m %>% filter(Coverage>8) %>% select(Path) %>% write_tsv("bamlists/44.bamlist", col_names = FALSE)
m %>% filter(Coverage>8) %>% select(NewPheno) %>% write_tsv("bamlists/44.pheno", col_names = FALSE)
m %>% filter(Coverage>8) %>% select(specimen_id) %>% write_tsv("bamlists/44.names", col_names = FALSE)
m44<-m %>% filter(Coverage>8)
meta<-m44

```

```{r}
ggplot(m44) + 
  geom_histogram(aes(x=Coverage,fill=Phenotype), alpha=0.75) +
  ylab("Count\n") +
  xlab("\nCoverage after Filtering") +
  theme_bw() +
  scale_fill_manual(values=c("red","blue"))
ggsave("outputs/104/coverage.jpeg")
```


# Run GWAS and call SNPs

```{r}
srun -t 72:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -P 12 \
 -doAsso 1 -yBin $HOME/barra-lh/bamlists/test47.pheno -GL 1 -minInd 42 -minMaf 0.10 \
 -minMapQ 20 -minQ 20 \
 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 -rf meta/lates-lgs.txt -out outputs/104/combined-asso-95 \
 -bam $HOME/barra-lh/bamlists/44.bamlist  > outputs/104/combined-asso.out 2> outputs/104/combined-asso.err &
```


```{sh, eval=FALSE}
srun -t 4-24:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 42 -bam bamlists/42.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf meta/lates-lgs.txt -out outputs/104/snps-wgs-05-glf  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doGLF 2 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/104/snps-wgs-05-glf.out 2> outputs/104/snps-wgs-05-glf.err &
```

1130415

```{sh, eval=FALSE}
plink --tped snps-wgs-05-glf.tped --tfam snps-wgs-05-glf.tfam  --out plink-binary-05 --recode --allow-extra-chr --noweb
plink --ped plink-binary-05.ped --map plink-binary-05.map --recode vcf --allow-extra-chr -out plink-05
bgzip plink-05.vcf 
tabix plink-05.vcf.gz

#renaming vcf bcftools/1.13
module load bcftools

bcftools reheader --samples bamlists/84.names -o outputs/103/renamed-05.vcf.gz outputs/103/plink-05.vcf.gz


bcftools +fill-tags outputs/103/renamed-05.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/103/pruned-05.vcf

bgzip outputs/103/pruned-05.vcf
tabix outputs/103/pruned-05.vcf.gz
```
