---
title: "105-alternative-fish"
output: html_document
date: "2025-09-05"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message=FALSE)
```


Include, Daly, Roper, FitzroyQLD

```{r}
qld<-read_csv("meta/qld-fish.csv") %>% mutate(NewPhenotype=Phenotype) %>% filter(Group!="Mainstream")
qld
```


```{r}
mq<-read_csv("~/github/barra-lh/meta/meta44.csv") %>% mutate(Group="Burdekin") %>% bind_rows(qld) %>% filter(Coverage > 8)
mq$Group<-factor(mq$Group, levels=c("Daly River","Roper River","Mainstream","Burdekin","FitzroyQLD"))
ggplot(mq) +
  geom_histogram(aes(x=Coverage, fill=Group)) +
  scale_fill_viridis_d(option="magma") +
  theme_bw() +
  ylab("Count\n") +
  xlab("\nCoverage") 
```


Should probably downsample these...


```{r}
mq %>% select(Path) %>% write_tsv("~/github/barra-lh/bamlists/69.bamlist", col_names = FALSE)
mq %>% select(Pheno) %>% write_tsv("~/github/barra-lh/bamlists/69.pheno", col_names = FALSE)
mq %>% select(specimen_id) %>% write_tsv("~/github/barra-lh/bamlists/69.names", col_names = FALSE)
meta<-mq
```



Call snps on all chroms

```{sh, eval=FALSE}
srun -t 5-24:00:00 -p high --mem=32G --nodes=1 --ntasks-per-node=1 --cpus-per-task=12 $HOME/angsd/angsd -nthreads 12 \
-minInd 66 -bam bamlists/69.bamlist -ref genome/GCF_001640805.2_TLL_Latcal_v3_genomic.fna \
-rf meta/lates-lgs.txt -out outputs/105/snps-wgs-05-glf  \
-minMaf 0.05 -minMapQ 20 -minQ 20 -GL 1 -doGLF 2 -doMajorMinor 1 -doMaf 1 -SNP_pval 1e-6 \
-doGeno 2 -doPost 1 -postCutoff 0.95 -doPlink 2  > outputs/105/snps-wgs-05-glf.out 2> outputs/105/snps-wgs-05-glf.err &
```

1130415

```{sh, eval=FALSE}
plink --tped snps-wgs-05-glf.tped --tfam snps-wgs-05-glf.tfam  --out plink-binary-05 --recode --allow-extra-chr --noweb
plink --ped plink-binary-05.ped --map plink-binary-05.map --recode vcf --allow-extra-chr -out plink-05
bgzip plink-05.vcf 
tabix plink-05.vcf.gz

#renaming vcf bcftools/1.13
module load bcftools

bcftools reheader --samples bamlists/84.names -o outputs/103/renamed-05.vcf.gz outputs/103/plink-05.vcf.gz


bcftools +fill-tags outputs/103/renamed-05.vcf.gz  -- -t MAF,F_MISSING | bcftools view -i 'F_MISSING < 0.05' | bcftools view -i 'INFO/MAF > 0.05' |  bcftools +prune -m 0.20 -w 50kb --nsites-per-win 1 --nsites-per-win-mode rand  > outputs/103/pruned-05.vcf

bgzip outputs/103/pruned-05.vcf
tabix outputs/103/pruned-05.vcf.gz
```
